### ISSUE TITLE
Add weight decay to neural network in artificial intelligence

### ISSUE THREAD

**Issue by EricAragorn** [OpenGenus/cosmos#2556]
**This is an:**

- [ ] New algorithm
- [x] Update to an existing algorithm
- [ ] Error
- [ ] Proposal to the Repository

**Details:**

Currently there is no weight decay or any kinds of weight regulation in neural_network.py
It would be helpful to add weight decay to generalize the data better

**Comment by kss682** [comment]
Hi, I would like to work on this issue.
Thanks
